mods/parsoid.js

```diff
@@ -530,6 +551,45 @@ function cheapBodyInnerHTML(html) {
     }
 }
 
+/**
+ * Replaces sections in original content with sections provided in sectionsJson
+ */
+function replaceSections(original, sectionsJson) {
+    var sectionOffsets = original['data-parsoid'].body.sectionOffsets,
+        newBody = cheapBodyInnerHTML(original.html.body);
+    Object.keys(sectionsJson)
+    .sort(function(id1, id2) {
+        var offset1 = sectionOffsets[id2],
+            offset2 = sectionOffsets[id1];
+        if (!offset1 || !offset2) {
+            throw new rbUtil.HTTPError({
+                status: 400,
+                body: {
+                    type: 'invalid_request',
+                    description: 'Invalid section ids'
+                }
+            });
+        }
+        return offset1.html[0] - offset2.html[0];
+    })
+    .forEach(function(id) {
+        var offset = sectionOffsets[id];
+        if (!offset) {
+            throw new rbUtil.HTTPError({
```

Comment:
@d00rman In case there's only 1 item in the sort comparator function wouldn't be called.